### PR TITLE
[New Feature][Ubuntu] Support autoinstall in ansible for ubuntu desktop

### PR DESCRIPTION
**Pull request by ZouYuhua**
changes

    support autoinstall in ansible for ubuntu desktop
    optimize autoinstall for Photon with ansilbe module iso_customize.py
    optimize autoinstall for Debian with ansilbe module iso_customize.py

Note:
replace the https://github.com/vmware/ansible-vsphere-gos-validation/pull/377 due to that the commit from orginal pull request 377  are not signed.

### CODE REVIEW

**Comment by ZouYuhua** [comment]
Replace the https://github.com/vmware/ansible-vsphere-gos-validation/pull/377 due to that the commits from orginal pull request 377 are not signed off.

**Comment by keirazhang** [comment]
Please update README.md:
1. Remove below prerequisite from README.md file. Thanks~
```
To test VMware Photon OS, Debian, and Ubuntu with ISO installation, it is also required to install xorriso package on your control machine.
```
2. Add Ubuntu desktop to supported guest OS: https://github.com/vmware/ansible-vsphere-gos-validation#supported-guest-os 

Thanks~

File: linux/deploy_vm/rebuild_unattend_install_iso.yml [hunk 1]
```diff
@@ -21,245 +21,265 @@
...
 
-# Set facts about files and command for creating an autoinstall ISO for Ubuntu live server
-- block:
-    - name: "Set fact of files to be update for Ubuntu unattended install"
-      ansible.builtin.set_fact:
-        iso_files_to_update:
-          - '/boot/grub/grub.cfg'
-          - '/md5sum.txt'
+    - name: "Search string in md5sum.txt"
+      ansible.builtin.shell: "cat {{ src_iso_file_dir }}/md5sum.txt | grep '/install\.386/'"
```
Comment (keirazhang) [inline linux/deploy_vm/rebuild_unattend_install_iso.yml:127]: You can directly use command
```
grep '/install\.386/' {{ src_iso_file_dir }}/md5sum.txt
```

Comment (ZouYuhua) [reply linux/deploy_vm/rebuild_unattend_install_iso.yml:127]: Thanks. Done

**Comment by ZouYuhua** [comment]
Testing is going.....

**Comment by ZouYuhua** [comment]
Test result:

1. Photon ISO:
    
image

2. Photon OVA:
image

3. ubuntu desktop with NVME+EFI:
    
image

4. ubuntu desktop with PVSCSI+BIOSI:
    
image


Note:
Running the testing for Debian 32bit and 64bit.

**Comment by ZouYuhua** [comment]
5. Debian 32-bit:
    
image

6. Debian 64-bit
image

**Comment by ZouYuhua** [comment]
5. Debian 32-bit:
    
image

7. Debian 64-bit:
    
image

File: README.md [hunk 2]
```diff
@@ -75,7 +74,7 @@ This project supports below scenarios for end-to-end guest OS validation testing
 | Photon OS 3.x, 4.x                              | :heavy_check_mark:               | :heavy_check_mark:       | :heavy_check_mark:                 |
 | Ubuntu 18.04 and later live-server              | :heavy_check_mark:               |                          | :heavy_check_mark:                 |
 | Ubuntu 20.04 and later cloud image              |                                  | :heavy_check_mark:       | :heavy_check_mark:                 |
-| Ubuntu 18.04 and later desktop                  |                                  |                          | :heavy_check_mark:                 |
+| Ubuntu 20.04 and later desktop                  | :heavy_check_mark:               |                          | :heavy_check_mark:                 |
```
Comment (Tomorrow9) [inline README.md:77]: How about "Ubuntu 18.04 desktop"?

Comment (ZouYuhua) [reply README.md:77]: I have not tested ubuntu 18.04 this time. So exclude it.  We only need to test update release for ubuntu 20.04, 22.04 and new releases in the future.

File: linux/deploy_vm/deploy_vm_from_iso.yml [hunk 3]
```diff
@@ -153,6 +153,17 @@
         - unattend_install_conf is defined
         - ('UOS' in unattend_install_conf)
 
+    # For ubuntu desktop, it's invalid with "echo '{{ autoinstall_complete_msg }}'' > /target/dev/ttyS0"
```
Comment (Tomorrow9) [inline linux/deploy_vm/deploy_vm_from_iso.yml:156]: This comment causes confusing, can be improved

Comment (ZouYuhua) [reply linux/deploy_vm/deploy_vm_from_iso.yml]: Thanks. Done

File: README.md (hunk 2 above)
Comment (ZouYuhua) [reply README.md:78]: Thanks. Done

**Review by Tomorrow9 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**